src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,29 @@
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { PostDetails } from '../PostData/PostDetails';
+import { StateContext } from '../../reducer/store';
+
+export const Sidebar = () => {
+  const { selectedPost } = useContext(StateContext);
+
+  return (
+    <div
+      data-cy="Sidebar"
+      className={classNames(
+        'tile',
+        'is-parent',
+        'is-8-desktop',
+        'Sidebar',
+        {
+          'Sidebar--open': selectedPost,
+        },
+      )}
+    >
+      {selectedPost && (
+        <div className="tile is-child box is-success ">
+          <PostDetails />
```

Comment:
In addition to the selected post data, you may also need to pass down functions as props to handle adding and deleting comments if these actions are supposed to be performed from the PostDetails component.